### ISSUE TITLE
Improve color contrast of row items

### ISSUE THREAD

**Issue by Jesse-Box** [getsentry/sentry#57285]
## Problems
This mainly focuses on addressing accessibility issues related to the row items found in the tabs of the replay details page. However, for practical reasons, I've include various design nits which relate to improving these rather stateful components 

### Poor Color Contrast
![Screenshot of Console Log](https://github.com/getsentry/sentry/assets/6508377/415b2aa5-322f-4272-8107-19e4bcc696ad)
- Using red or yellow text in the console tab violates the minimum requirements
- Having muted gray content to symbolise a row that has yet to have been played violates minimum requirements

---

### Poor Signifiers for Selectable Rows:
![Screenshot of Network Request being hovered](https://github.com/getsentry/sentry/assets/6508377/fc209960-219b-45ef-893c-93ff3570c1e9)
For selectable row items such as network requests and accessibility violations, there aren't many signifiers to indicate that you can click to see more details.

---

### Visual Inconsistencies
![Screenshot of Network Request](https://github.com/getsentry/sentry/assets/6508377/2abc15f1-2740-4660-b9b8-aa6418242978)
![Screenshot of Console Log](https://github.com/getsentry/sentry/assets/6508377/f7ce09d5-1eb3-46a3-a260-74ea8dd4ecbe)
![Screenshot of Breadcrumb](https://github.com/getsentry/sentry/assets/6508377/68a04598-1dde-4b39-8d3d-ac4c0e0343d2)
DOM (Now Breadcrumb) items don't tint errors, but console and network do

---

### Excessive padding
- Y-Paddings: The minimum height of a row item can be lowered, giving the user more information
- X-Paddings: Each Cell inside a network request and error row has their padding, reducing the amount of information visible

---

### Acceptance Criteria
1. Default text color is gray400, icons can still be coloured according to their severity
2. We no longer distinguish between rows that are elapsed, current or pending. Now only if current or not.
3. Selectable row items get a hover and pressed colour and these colours change when the row item is selected or of a specific variant (default vs error)2. If a row item as an error or warning variant, its background gets tinted in their respective colors
3. Reduce row item Y-paddings where possible
4. Reduce row item X-paddings where possible

### Designs
[Figma](https://www.figma.com/file/tlGOyIijfu309du7zXTDTM/Library%3A-Replay?type=design&node-id=638%3A1133&mode=design&t=lNkRVNK6EUbyEmC4-1)

### Related To
#57273

**Comment by Jesse-Box** [comment]
@ryan953 I've added some additional information to this ticket to make my requests a bit clearer
